### DISCUSSION TITLE
@cypress/react and react-helmet. Can they work together?

### DISCUSSION THREAD

**Discussion by glomotion** [cypress-io/cypress#16265] [Component Testing]
So the cypress docs mention things like `cy.title()` to verify that the `document.title` is updating as things in react trigger title changes via react-helmet...

Sadly though, when @cypress/react components are being tested (eg inside cypress run-ct) `cy.title()` ALWAYS returns the same string: `Components App`.

Has anyone managed to get `@cypress/react` & `react-helmet` working together (so that helmet actually updates cy.title, for example)?

**Comment by glomotion** [comment]
FWIW, we tried to catch helmet updates from within the fixture, but no joy (see screenshots).

![Screen Shot 2021-04-29 at 2 08 58 pm](https://user-images.githubusercontent.com/1452237/116501611-9928b300-a8f4-11eb-82d5-6cf5f7d75fe5.png)
![Screen Shot 2021-04-29 at 2 09 07 pm](https://user-images.githubusercontent.com/1452237/116501617-9b8b0d00-a8f4-11eb-9e63-ce00f99ab73b.png)

☝️ InventoryPage component triggers a helmet title change immediately on-mount.

**Comment by bahmutov** [comment]
I think because the helmet is really meant to operate at the top level - at the document level, a more appropriate test here would be an E2E test, not a component test.
